gradio/blocks.py

```diff
@@ -254,9 +254,16 @@ def move_resource_to_block_cache(
         else:
             url_or_file_path = str(utils.abspath(url_or_file_path))
             if not utils.is_in_or_equal(url_or_file_path, self.GRADIO_CACHE):
-                temp_file_path = processing_utils.save_file_to_cache(
-                    url_or_file_path, cache_dir=self.GRADIO_CACHE
-                )
+                try:
```

Comment:
This makes sense but I think we're not covering the case where the value of the component is a file on the space.

I loaded the following demo

```
gr.load("gradio/image_mod_default_image", src="spaces").launch()
```

And this is what I see

<img width="829" alt="image" src="https://github.com/gradio-app/gradio/assets/41651716/1d0fcc02-ab21-4b2e-900b-4ab67d3c20d0">


From the console

```
file-url.ts:12 Uncaught (in promise) TypeError: Failed to construct 'URL': Invalid URL
    at should_proxy_wasm_src (file-url.ts:12:14)
    at resolve_wasm_src (file-url.ts:26:22)
    at $$self.$$.update (Image.svelte:28:3)
    at init (svelte.js:3246:5)
    at new Image (Image.svelte:30:21)
    at create_if_block (Example-9a0e0dba.js:29:10)
    at create_fragment (Example.svelte:19:39)
    at init (svelte.js:3250:34)
    at new Example (Example.svelte:8:29)
    at construct_svelte_component (svelte.js:1613:9)
```

and 

```
GET http://127.0.0.1:7861/proxy=https://gradio-image-mod-default-image.hf.space/file=undefined 403 (Forbidden)
```

Reply:
Hmm okay I'll look at this more closely on Monday, thanks for catching this @freddyaboulton

Reply:
I actually can't repro this issue, the Space loads for me:

<img width="1250" alt="image" src="https://github.com/gradio-app/gradio/assets/1778297/b123366f-27d5-4545-b961-8ec2c490e9a5">

I do have an issue when actually trying to run the app, but that also will be tackled as part of https://github.com/gradio-app/gradio/issues/7360

Reply:
I don't think this is related to #7360 actually.

The function server returns the right path but the issue is that the `FileData` object is wrapped in a list - 

<img width="700" alt="image" src="https://github.com/gradio-app/gradio/assets/41651716/9fc79d7c-ba22-4f0a-8d65-9e62fd4d0fdb">

I think this may be because of the change to `reduce_singleton_output`?

Reply:
I'll take a look!